### ISSUE TITLE
New-Object : Exception calling ".ctor" with "1" argument(s): "Error MediaInfo_Open"

### ISSUE THREAD

**Issue by baal77** [stax76/Get-MediaInfo#1]
Hello Stax76,

When launching Get-MediaInfo on a single file in the current folder, I get this error.

```
Downloads> Get-MediaInfo '.\Waves - 5121.mp4'
New-Object : Exception calling ".ctor" with "1" argument(s): "Error MediaInfo_Open"
At C:\Users\b77\OneDrive\documents\powershell\includes\Get-MediaInfo\Get-MediaInfo.ps1:109 char:27
+                     $mi = New-Object MediaInfo -ArgumentList $File
+                           ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : InvalidOperation: (:) [New-Object], MethodInvocationException
    + FullyQualifiedErrorId : ConstructorInvokedThrowException,Microsoft.PowerShell.Commands.NewObjectCommand
```

I've noticed that when I'm in the parent directory and query the same file, it works.

```
b77> Get-MediaInfo '.\Downloads\Waves - 5121.mp4'


FileName       : Waves - 5121.mp4
Ext            : mp4
Format         : AVC
DAR            : 1.778
Width          : 1920
Height         : 1080
BitRate        : 3844.199
Duration       : 0.16783333333333333
FileSize       : 4.6124143600463867
FrameRate      : 30
ScanType       : Progressive
ColorPrimaries :
Transfer       :
FormatProfile  : High@L4
AudioCodec     : AAC LC
TextFormat     :
Directory      : .\Downloads
```

It seems that your functions doesn't take in account, as input, a file into the current folder.
I'm not sure to have a real explanation (:-D),  but it seems that if you add this line to provide the full path instead of the relative path at the top begin of your "Process" block in each function, everything works perfectly, for files, or folders.
`$Path = (Resolve-Path $Path).Path`

Not sure if it's the good way to correct this, but it works for me.

To finish, I just want to thank you, A LOT, for Staxrip, this Get-Media script and for your mpv.net that I just discovered yesterday ... It's incredibly fast, light, and full of customization.
It's perfect, and better than VLC at my eyes.
THANKS for all your softwares.

Fred

**Comment by stax76** [comment]
Hello Fred,

for me it's not reproducible:

```
Samples> Get-MediaInfo '.\Big Buck Bunny.mkv'


FileName       : Big Buck Bunny.mkv
Ext            : mkv
Format         : VP9
DAR            : 1,778
Width          : 3840
Height         : 2160
BitRate        : 0
Duration       : 10,5777666666667
FileSize       : 1601,1530456543
FrameRate      : 60
ScanType       :
ColorPrimaries : BT.709
Transfer       : BT.709
FormatProfile  :
AudioCodec     : AAC LC
TextFormat     :
Directory      : .
```

There might be something different on your system. Looking at the powershell and C# code it looks like the path is passed as is to MediaInfo so MediaInfo receives the relative path exactly like you passed it in. I assume MediaInfo uses some standard WinAPI function to open it and standard WinAPI functions support relative paths, here it works and it's very strange that it does not work on your environment. What are your specs?

I'm using:

Windows 10 1909
PowerShell 5.1
Windows Terminal 0.11.1191.0

Here it works with and without Windows Terminal.

Or maybe it has something to do with the way you installed the script or how your system is configured.

Does `Test-Path` return true?

Maybe it has something to do with OneDrive.

**Comment by baal77** [comment]
Hello Stax76,

Sorry for the delay in my answer.
Here are some additional details concerning the environment of my computer :
```
b77> $PSVersionTable

Name                           Value
----                           -----
PSVersion                      5.1.18362.752
PSEdition                      Desktop
PSCompatibleVersions           {1.0, 2.0, 3.0, 4.0...}
BuildVersion                   10.0.18362.752
CLRVersion                     4.0.30319.42000
WSManStackVersion              3.0
PSRemotingProtocolVersion      2.3
SerializationVersion           1.1.0.1
```

```
b77> [System.Environment]::OSVersion.Version

Major  Minor  Build  Revision
-----  -----  -----  --------
10     0      18363  0
```
It's a Windows 10 Pro version 1909 1836.815.

Concerning the result of `Test-Path`, I'm not sure to have well understood what you exactly want to test, so I did the 3 following tests :

```
Downloads> Test-Path
cmdlet Test-Path at command pipeline position 1
Supply values for the following parameters:
Path[0]:

Downloads> Test-Path .
True

Downloads> Test-Path '.\Waves - 5121.mp4'
True
```

The results are the same, whatever terminal I use :
- Native PowerShell Console
- Native PowerShell ISE
- Windows Terminal Version: 0.11.1191.0
```
b77> $host


Name             : ConsoleHost
Version          : 5.1.18362.752
InstanceId       : 27459dd5-04c7-4ba1-830d-a2c9cb7cfd3f
UI               : System.Management.Automation.Internal.Host.InternalHostUserInterface
CurrentCulture   : en-IE
CurrentUICulture : en-GB
PrivateData      : Microsoft.PowerShell.ConsoleHost+ConsoleColorProxy
DebuggerEnabled  : True
IsRunspacePushed : False
Runspace         : System.Management.Automation.Runspaces.LocalRunspace
```
- Visual Studio code 
```
Name             : Visual Studio Code Host
Version          : 2020.4.0
InstanceId       : 67fb420e-0394-44f9-927d-f72a52a7adb7
UI               : System.Management.Automation.Internal.Host.InternalHostUserInterface
CurrentCulture   : en-IE
CurrentUICulture : en-GB
PrivateData      : Microsoft.PowerShell.EditorServices.Services.PowerShellContext.EditorServicesPSHost+ConsoleColorProxy
DebuggerEnabled  : True
IsRunspacePushed : False
Runspace         : System.Management.Automation.Runspaces.LocalRunspace
```

I also though about a problem linked to OneDrive after I've sent my precedent message, so I did a new test right now, on another computer on which I've not OneDrive installed at all.
It's also a computer on which I never played with PowerShell before, so, there is no customization at all, anywhere (it's a French version, but should not have any impact).

So, to avoid any doubt, I downloaded Get-MediaInfo from GitHub, and put it on my Desktop, then I create a profile.ps1 in C:\Windows\System32\WindowsPowerShell\v1.0 (I've not other profile elsewhere) and only put ". C:\Users\Fred\Desktop\Get-MediaInfo\Get-MediaInfo.ps1" into it.

Result are the same.
Find them here, and the details of this other environment :
```
PS C:\Users\Fred\Downloads> Get-MediaInfo .\VIDEO-2019-01-13-17-43-5911.mp4
New-Object : Exception lors de l'appel de « .ctor » avec « 1 » argument(s) : « Error MediaInfo_Open »
Au caractère C:\Users\Fred\Desktop\Get-MediaInfo\Get-MediaInfo.ps1:107 : 27
+                     $mi = New-Object MediaInfo -ArgumentList $File
+                           ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : InvalidOperation : (:) [New-Object], MethodInvocationException
    + FullyQualifiedErrorId : ConstructorInvokedThrowException,Microsoft.PowerShell.Commands.NewObjectCommand
```

```
PS C:\Users\Fred\Downloads> $PSVersionTable

Name                           Value
----                           -----
PSVersion                      5.1.18362.752
PSEdition                      Desktop
PSCompatibleVersions           {1.0, 2.0, 3.0, 4.0...}
BuildVersion                   10.0.18362.752
CLRVersion                     4.0.30319.42000
WSManStackVersion              3.0
PSRemotingProtocolVersion      2.3
SerializationVersion           1.1.0.1
```

```
PS C:\Users\Fred\Downloads> [System.Environment]::OSVersion.Version

Major  Minor  Build  Revision
-----  -----  -----  --------
10     0      18362  0
```
It's a Windows 10 Home version 1903 18362.778.

```
PS C:\Users\Fred\Downloads> $host


Name             : ConsoleHost
Version          : 5.1.18362.752
InstanceId       : d1cb5207-b984-4304-a9f7-a2a3b6d54284
UI               : System.Management.Automation.Internal.Host.InternalHostUserInterface
CurrentCulture   : fr-FR
CurrentUICulture : fr-FR
PrivateData      : Microsoft.PowerShell.ConsoleHost+ConsoleColorProxy
DebuggerEnabled  : True
IsRunspacePushed : False
Runspace         : System.Management.Automation.Runspaces.LocalRunspace
```

**Comment by stax76** [comment]
Thanks for the effort Fred, it might be too difficult to find out how to reproduce this, it doesn't seem using Resolve-Path has any disadvantage, in fact I think it's probably better anyway, so I just applied it everywhere and made a quick test to verify everything is working. And I've improved the way the native mediainfo.dll is found, it no longer relies on modifying the path environment variable of the process. I've also locally updated and tested MediaInfo.NET and StaxRip.

v3.1 is online
